gslib/utils/ls_helper.py

```diff
@@ -371,12 +371,15 @@ def ExpandUrlAndPrint(self, url):
       return self._RecurseExpandUrlAndPrint(url.url_string,
                                             print_initial_newline=False)
     else:
-      # User provided a prefix or object URL, but it's impossible to tell
-      # which until we do a listing and see what matches.
+      # User provided a prefix or object URL, but can't tell which unless there
+      # is a generation or we do a listing to see what matches.
+      if url.HasGeneration():
```

Comment:
Do you know if cp has the same issue? If yes then we might have to move this logic to wildcard iterator.

Reply:
It doesn't seem to if this is what you mean:
```
$ gsutil ls -a gs://0noogler-kingdom
gs://0noogler-kingdom/foo1#1630600624018517
gs://0noogler-kingdom/foo1#1630674094655571
gs://0noogler-kingdom/wild*card/
$ gsutil cp gs://0noogler-kingdom/foo1#1630674094655571 gs://0noogler-kingdom/foo2
Copying gs://0noogler-kingdom/foo1#1630674094655571 [Content-Type=application/octet-stream]...
/ [1 files][    0.0 B/    0.0 B]                                                
Operation completed over 1 objects.
```

Reply:
Yep, that's what I mean. Thanks for checking.